### PR TITLE
feature: Improve Dell OpenManage Discovery

### PR DESCRIPTION
**Pull request by Rosiak**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
screen shot 2016-11-08 at 22 45 45

### CODE REVIEW

File: includes/discovery/sensors/fanspeeds/dell.inc.php [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-            $fan_oid         = ".1.3.6.1.4.1.674.10892.1.700.12.1.6.1.$index";
-            $descr_oid       = "coolingDeviceLocationName.1.$index";
-            $limit_oid       = "coolingDeviceLowerCriticalThreshold.1.$index";
-            $descr           = trim(snmp_get($device, $descr_oid, '-Oqv', 'MIB-Dell-10892'), '"');
-            $descr           = preg_replace('/(Board | MOD )/', '', $descr);
-            $current         = snmp_get($device, $fan_oid, '-Oqv', 'MIB-Dell-10892');
-            $low_limit       = snmp_get($device, $limit_oid, '-Oqv', 'MIB-Dell-10892');
-            $divisor         = '1';
-            discover_sensor($valid['sensor'], 'fanspeed', $device, $fan_oid, $index, 'dell', $descr, $divisor, '1', $low_limit, null, null, null, $current);
+            discover_sensor($valid['sensor'], 'fanspeed', $device, $cur_oid . $index, $index, 'dell', $descr, '0', '1', $lowlimit, $low_warn_limit, $warnlimit, $limit, $value, 'snmp', $index);
```
Comment (laf) [inline includes/discovery/sensors/fanspeeds/dell.inc.php]: Does the OID change here as before we had `.1.3.6.1.4.1.674.10892.1.700.12.1.6.1.` + the index now we have `.1.3.6.1.4.1.674.10892.1.700.12.1.6.` + the index.

File: includes/discovery/sensors/states/dell.inc.php [hunk 2]
```diff
@@ -1,98 +1,145 @@
...
+                        $descr = $tablevalue[3];
+                    } elseif ($state_name == 'batteryState') {
+                        $descr = str_replace('"', "", snmp_get($device, "batteryConnectionControllerName." .$index. "", "-Ovqn", $tablevalue[4])) . ' - ' .$temp[$index][$tablevalue[3]];
+                    } elseif ($state_name == 'arrayDiskState') {
+                        $descr = str_replace('"', "", snmp_get($device, "arrayDiskEnclosureConnectionEnclosureName." .$index. "", "-Ovqn", $tablevalue[4])) . ' - ' .$temp[$index][$tablevalue[3]];
+                    } else {
+                        $descr = clean($temp[$index][$tablevalue[3]]); // Use clean as virtualDiskDeviceName is user defined
+                    }
+                    //Discover Sensors
+                    discover_sensor($valid['sensor'], 'state', $device, $cur_oid.$index, $index, $state_name, $descr, '1', '1', null, null, null, null, $temp[$index][$tablevalue[2]], 'snmp', $index);
```
Comment (laf) [inline includes/discovery/sensors/states/dell.inc.php]: Do any of the OID's change here?

File: includes/discovery/sensors/temperatures/dell.inc.php [hunk 3]
```diff
@@ -1,64 +1,30 @@
...
-                $warnlimit    = snmp_get($device, $warnlimit_oid, '-Oqv', 'MIB-Dell-10892');
-                $limit        = snmp_get($device, $limit_oid, '-Oqv', 'MIB-Dell-10892');
-                $lowlimit     = snmp_get($device, $lowlimit_oid, '-Oqv', 'MIB-Dell-10892');
-
-                discover_sensor($valid['sensor'], 'temperature', $device, $temperature_oid, $temperature_id, 'dell', $descr, '10', '1', ($lowlimit / 10), ($low_warn_limit / 10), ($warnlimit / 10), ($limit / 10), ($temperature / 10));
-            }
-        }//end if
-    }//end foreach
-}//end if
+            discover_sensor($valid['sensor'], 'temperature', $device, $cur_oid . $index, $index, 'dell', $descr, $divisor, '1', $lowlimit, $low_warn_limit, $warnlimit, $limit, $value, 'snmp', $index);
```
Comment (laf) [inline includes/discovery/sensors/temperatures/dell.inc.php]: Same here.

File: includes/discovery/sensors/fanspeeds/dell.inc.php (hunk 1 above)
Comment (Rosiak) [inline includes/discovery/sensors/fanspeeds/dell.inc.php]: Nope OID stays the same.

File: includes/discovery/sensors/states/dell.inc.php (hunk 2 above)
Comment (Rosiak) [inline includes/discovery/sensors/states/dell.inc.php]: Nope OID stays the same.

File: includes/discovery/sensors/temperatures/dell.inc.php (hunk 3 above)
Comment (Rosiak) [inline includes/discovery/sensors/temperatures/dell.inc.php]: Nope OID stays the same.

File: includes/discovery/sensors/fanspeeds/dell.inc.php (hunk 1 above)
Comment (Rosiak) [inline includes/discovery/sensors/fanspeeds/dell.inc.php]: laf: Pretty sure.
```
coolingDeviceIndex.1.1 = 1
coolingDeviceIndex.1.2 = 2
coolingDeviceIndex.1.3 = 3
coolingDeviceIndex.1.4 = 4
coolingDeviceIndex.1.5 = 5
coolingDeviceIndex.1.6 = 6
coolingDeviceIndex.1.7 = 7
coolingDeviceIndex.1.8 = 8
coolingDeviceIndex.1.9 = 9
coolingDeviceIndex.1.10 = 10
coolingDeviceIndex.1.11 = 11
coolingDeviceIndex.1.12 = 12

Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.1, 1.1, dell, System Board FAN MOD 1A RPM, snmp, , 1.1
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.1'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.1'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.2, 1.2, dell, System Board FAN MOD 2A RPM, snmp, , 1.2
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.2'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.2'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.3, 1.3, dell, System Board FAN MOD 3A RPM, snmp, , 1.3
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.3'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.3'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.4, 1.4, dell, System Board FAN MOD 4A RPM, snmp, , 1.4
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.4'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.4'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.5, 1.5, dell, System Board FAN MOD 5A RPM, snmp, , 1.5
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.5'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.5'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.6, 1.6, dell, System Board FAN MOD 6A RPM, snmp, , 1.6
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.6'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.6'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.7, 1.7, dell, System Board FAN MOD 1B RPM, snmp, , 1.7
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.7'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.7'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.8, 1.8, dell, System Board FAN MOD 2B RPM, snmp, , 1.8
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.8'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.8'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.9, 1.9, dell, System Board FAN MOD 3B RPM, snmp, , 1.9
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.9'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.9'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.10, 1.10, dell, System Board FAN MOD 4B RPM, snmp, , 1.10
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.10'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.10'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.11, 1.11, dell, System Board FAN MOD 5B RPM, snmp, , 1.11
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.11'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.11'] 
.Discover sensor: .1.3.6.1.4.1.674.10892.1.700.12.1.6.1.12, 1.12, dell, System Board FAN MOD 6B RPM, snmp, , 1.12
SQL[SELECT COUNT(sensor_id) FROM `sensors` WHERE `poller_type`= 'snmp' AND `sensor_class` = 'fanspeed' AND `device_id` = '31' AND sensor_type = 'dell' AND `sensor_index` = '1.12'] 
SQL[SELECT * FROM `sensors` WHERE `sensor_class` = 'fanspeed' AND `device_id` = '31' AND `sensor_type` = 'dell' AND `sensor_index` = '1.12'] 
```

**Pull request merged**